src/Data/NatSet.agda

```diff
@@ -0,0 +1,73 @@
+{-# OPTIONS --without-K --safe #-}
+
+--------------------------------------------------------------------------------
+-- Simple implementation of sets of ℕ.
+--------------------------------------------------------------------------------
+
+module Data.NatSet where
```

Comment:
This looks like it could be implemented over any type with a suitable `StrictTotalOrder` right? I'd be tempted to move this to generalise it to `Data.List.Sets` to mirror the existing `Data.AVL.Sets`

Reply:
I don't think so—it is really centered around the type of the `compare` function (i.e. you have to be able to decompose the thing you want to compare). I suppose you could generalise it to some kind of splittable ordered monoid, but I don't really know a good way to do that, or a good algebra for it.

The reason for it is that it's (slightly) more efficient than just doing an ordered list and comparing on the Nats (I've added an explanation in the file about it).

I don't know how it compares to `Data.AVL`, though. I don't know how `Data.AVL`'s performance works with regards to Nats, i.e. does comparison use the machine-implemented O(1) version? Or does it use the unary representation? If it's the former, then it's probably a better idea to just use `Data.AVL`, if it's the latter this should be significantly faster.